### PR TITLE
Migrates Gateway code to MQTT example

### PR DESCRIPTION
**Pull request by gguuss**

### CODE REVIEW

**Comment by gguuss** [comment]
@hongalex PTAL, this migrates the code and tests from beta-features.

**Comment by hongalex** [comment]
Currently failing tests (though seems unrelated to our features)

**Comment by gguuss** [comment]
I'll resync and maybe it will jiggle good.

File: iot/api-client/mqtt_example/cloudiot_mqtt_example.py [hunk 1]
```diff
@@ -169,37 +169,200 @@ def get_client(
...
+    detach_topic = '/devices/{}/detach'.format(device_id)
+    print('Detaching: {}'.format(detach_topic))
+    client.loop()
+    client.connect(mqtt_bridge_hostname, mqtt_bridge_port)
+    client.publish(detach_topic, '{}', qos=1)
+    time.sleep(5)  # wait for the server to respond / will trigger callback
+    # [END detach_device]
+
+
+def attach_device(client, device_id, mqtt_bridge_hostname, mqtt_bridge_port):
```
Comment (hongalex) [inline iot/api-client/mqtt_example/cloudiot_mqtt_example.py]: `mqtt_bridge_*` variables seem unused here. Seems like `attach` and `detach` are not symmetrical, though I'm implementing it as symmetrical in other languages.

File: iot/api-client/mqtt_example/cloudiot_mqtt_example.py [hunk 2]
```diff
@@ -169,37 +169,200 @@ def get_client(
...
+    """Attach the device to the gateway."""
+    # [START attach_device]
+    attach_topic = '/devices/{}/attach'.format(device_id)
+    print('Attaching: {}'.format(attach_topic))
+    # TODO {'authorization': '<JWT_TOKEN>'}
```
Comment (hongalex) [inline iot/api-client/mqtt_example/cloudiot_mqtt_example.py]: Perhaps add a JWT string as a parameter so we can add this properly. I did this for Go

File: iot/api-client/mqtt_example/cloudiot_mqtt_example.py [hunk 3]
```diff
@@ -169,37 +169,200 @@ def get_client(
...
+    # [END attach_device]
+
+
+def listen_for_messages(
+        service_account_json, project_id, cloud_region, registry_id, device_id,
+        gateway_id, num_messages, private_key_file, algorithm, ca_certs,
+        mqtt_bridge_hostname, mqtt_bridge_port, jwt_expires_minutes, duration,
+        cb=None):
+    """Listens for messages sent to the gateway and bound devices."""
+    # [START listen_for_config_messages]
```
Comment (hongalex) [inline iot/api-client/mqtt_example/cloudiot_mqtt_example.py]: change region tag for this?

File: iot/api-client/mqtt_example/cloudiot_mqtt_example.py [hunk 4]
```diff
@@ -169,37 +169,200 @@ def get_client(
...
+                project_id, cloud_region, registry_id, gateway_id,
+                private_key_file, algorithm, ca_certs, mqtt_bridge_hostname,
+                mqtt_bridge_port)
+
+        time.sleep(1)
+
+    detach_device(client, device_id, mqtt_bridge_hostname, mqtt_bridge_port)
+
+    print('Finished.')
+    # [END listen_for_config_messages]
```
Comment (hongalex) [inline iot/api-client/mqtt_example/cloudiot_mqtt_example.py]: change the closing tag as well

File: iot/api-client/mqtt_example/cloudiot_mqtt_example.py [hunk 5]
```diff
@@ -169,37 +169,200 @@ def get_client(
...
+        project_id, cloud_region, registry_id, gateway_id,
+        private_key_file, algorithm, ca_certs, mqtt_bridge_hostname,
+        mqtt_bridge_port)
+
+    attach_device(client, device_id, mqtt_bridge_hostname, mqtt_bridge_port)
+    print('Waiting for device to attach.')
+    time.sleep(5)
+
+    # Publish state to gateway topic
+    gateway_state = 'Starting HUB at: {}'.format(time.time())
```
Comment (hongalex) [inline iot/api-client/mqtt_example/cloudiot_mqtt_example.py]: might be confusing if we use both Hub and Gateway. Might want to just change this to 'Starting gateway at...'

File: iot/api-client/mqtt_example/cloudiot_mqtt_example.py (hunk 1 above)
Comment (gguuss) [reply iot/api-client/mqtt_example/cloudiot_mqtt_example.py]: I wanted for these to be usable as stand-alone functions, if we don't want the connect call, lmk, the client can disconnect on error before the attach call.

File: iot/api-client/mqtt_example/cloudiot_mqtt_example.py (hunk 3 above)
Comment (gguuss) [reply iot/api-client/mqtt_example/cloudiot_mqtt_example.py]: done

File: iot/api-client/mqtt_example/cloudiot_mqtt_example.py (hunk 5 above)
Comment (gguuss) [reply iot/api-client/mqtt_example/cloudiot_mqtt_example.py]: done

**Comment by gguuss** [comment]
@hongalex Got the tests passing and added parameter for authorization JWT string to attach.

**Pull request merged**